### ISSUE TITLE
[BUG] linter fails when env variable is non-string

### ISSUE THREAD

**Issue by nirav24** [stackrox/kube-linter#136]
**System info:**
 - OS: MaxOS

**Describe the bug**
k8s requires env variable to be of type string. when non-string value is set as env, kube-linter fails with the below error
```
Warning: no valid objects found.
```

**To Reproduce**
- put below yaml under **dummy** dir
- run `kube-linter lint ./dummy`

**Sample YAML input**
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test
  labels:
    label1: content
spec:
  replicas: 1
  selector:
    matchLabels:
      label1: content
  template:
    metadata:
      labels:
        label1: content
    spec:
      containers:
        - name: test-container
          image: "image/image:tag"
          env:
          - name: TEST
            value: 1
```
**Expected behavior**
Linter should give a proper error message something like 

**Only type "string" is allowed.**

**Comment by kreamkorokke** [comment]
Could you try running it again with `-v` option? We understand that currently the error output with `-v` might not be the prettiest, but as of now it should at least give you some better hints than just the above error message you pointed out.

**Comment by nirav24** [comment]
> Could you try running it again with `-v` option? We understand that currently the error output with `-v` might not be the prettiest, but as of now it should at least give you some better hints than just the above error message you pointed out.

```
Warning: failed to load object from dummytests/deployments.yaml: failed to decode: v1.Deployment.Spec: 
v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Name: 
Image: Env: []v1.EnvVar: v1.EnvVar.Value: ReadString: expects " or n, but found 1, error found in #10 byte of 
...|,"value":1}],"image"|..., bigger context ...|lse,"containers":[{"env":
[{"name":"TEST","value":1}],"image":"image/image:tag","name":"kube-linter-t|...
Warning: no valid objects found.
```

**Comment by viswajithiii** [comment]
Hi @nirav24, this is a good point and is currently tracked in #63. Definitely something we need to fix. I'm closing this issue as a duplicate, but thanks for raising!
